File: retrofit_lib/src/main/java/com/wzgiceman/rxretrofitlibrary/retrofit_rx/utils/AppDataBase.java
package com.wzgiceman.rxretrofitlibrary.retrofit_rx.utils;

import android.arch.persistence.db.SupportSQLiteDatabase;
import android.arch.persistence.room.Database;
import android.arch.persistence.room.Room;
import android.arch.persistence.room.RoomDatabase;
import android.arch.persistence.room.migration.Migration;
import android.support.annotation.NonNull;

import com.wzgiceman.rxretrofitlibrary.retrofit_rx.RxRetrofitApp;
import com.wzgiceman.rxretrofitlibrary.retrofit_rx.download.DownInfoEntity;
import com.wzgiceman.rxretrofitlibrary.retrofit_rx.http.cookie.CookieResulte;

@Database(entities = {DownInfoEntity.class , CookieResulte.class}, version = 2,exportSchema = false)
public abstract class AppDataBase extends RoomDatabase {

    private volatile static AppDataBase dataBase;

    public static final String DATABASE_NAME = "retrofitLib";

    private static final String SQL_FORMAT = "(%d,'"+ "%s"  +"'),";

    public static AppDataBase getInstance() {

        synchronized (AppDataBase.class){

            if(dataBase==null){

                dataBase =
                        Room.databaseBuilder(RxRetrofitApp.getApplication(),AppDataBase.class,DATABASE_NAME)
                                .addCallback(new Callback() {
                                    @Override
                                    public void onCreate(@NonNull SupportSQLiteDatabase db) {
                                        super.onCreate(db);
                                    }

                                    @Override
                                    public void onOpen(@NonNull SupportSQLiteDatabase db) {
                                        super.onOpen(db);
                                    }
                                }).allowMainThreadQueries()
                                .setJournalMode(JournalMode.TRUNCATE)
                                .build();
            }
        }
        return dataBase;
    }

    public abstract DownLoadDao getDownloadDao();

    public abstract CookieDao getCookieDao();
}

File: roomdemo/src/main/java/com/example/roomdemo/AppDataBase.kt
package com.example.roomdemo

import android.arch.persistence.room.Database
import android.arch.persistence.room.Room
import android.arch.persistence.room.RoomDatabase
import android.content.Context
import com.example.roomdemo.dao.UserDao
import com.example.roomdemo.entity.User

@Database(entities = [User::class],version = 1,exportSchema = false)
abstract class AppDataBase :RoomDatabase(){

    companion object{  //构建单例对象
        private const val DB_NAME = "appDatabase"
        @Volatile
        private var INSTANCE:AppDataBase ?=null

        fun getDatabase(context: Context):AppDataBase{
            val temInstance= INSTANCE
            if(temInstance !=null){ return temInstance}
            synchronized(this){
               return Room.databaseBuilder(context.applicationContext,
                        AppDataBase::class.java, DB_NAME)
                        .allowMainThreadQueries()
                        .build()
            }
        }
    }
    //获取数据表操作实例
    abstract fun userDao():UserDao
}